dbms/src/Interpreters/Context.cpp

```diff
@@ -1850,6 +1855,17 @@ void Context::initializeSystemLogs()
     shared->system_logs.emplace(*global_context, getConfigRef());
 }
 
+void Context::createSystemLogs()
+{
+    auto lock = getLock();
+    shared->system_logs->query_log->prepareTable();
```

Comment:
`shared->system_logs->*_log` is `nullptr` if the log is not enabled in config (see `SystemLogs::SystemLogs(...)`)

Reply:
Rewrote to if (table) {do stuff}